x-pack/legacy/plugins/uptime/public/apps/index.ts

```diff
@@ -4,7 +4,8 @@
  * you may not use this file except in compliance with the Elastic License.
  */
 
-import { compose } from '../lib/compose/kibana_compose';
-import { startApp } from './start_app';
+import chrome from 'ui/chrome';
+import { npStart } from 'ui/new_platform';
+import { Plugin } from './plugin';
 
-startApp(compose());
+new Plugin(npStart, chrome).start();
```

Comment:
In the NP the class will receive a `PluginInitializerContext` in it's constructor and `CoreStart` in it's `.start()` method:

`new Plugin(pluginInitializerContext).start(npStart)`

If you don't need any API's from `PluginInitializerContext` you can just use an empty object. You can keep the second constructor argument for injecting any legacy dependencies that will eventually be removed / replaced by a NP equivalent. So then it will become:

```ts
new Plugin({}, chrome).start(npStart);
```

Reply:
I believe this is resolved now in https://github.com/elastic/kibana/pull/44722/commits/638b589c94fd65bbb54ba08c016828508110ae00.